libr/debug/p/native/linux/linux_debug.c

```diff
@@ -172,7 +175,7 @@ static void linux_remove_fork_bps(RDebug *dbg) {
 	// Set dbg tid to the new child temporarily
 	dbg->pid = dbg->forked_pid;
 	dbg->tid = dbg->forked_pid;
-	r_debug_select (dbg, dbg->forked_pid, dbg->forked_pid);
+	// r_debug_select (dbg, dbg->forked_pid, dbg->forked_pid);
```

Comment:
This should be removed because it will call `linux_attach_new_process` which calls `linux_detach_all` before attaching to the forked process. This is problematic because `PTRACE_DETACH` will continue the parent process before removing its breakpoints and the parent process might terminate before we have removed all breakpoints in the forked process and return to it.

The best way to get access to registers and memory is to temporarily change the `dbg->pid` and `dbg->tid`, so calls like `r_debug_reg_sync` can succeed.

Reply:
Please run a test program with a fork and a breakpoint after the fork, will child will simply die. This still has to be solved.

If you don't select the forked process, the code will still run and remove breakpoints from the parent that shouldn't be removed, I am not sure you understood this function. If debug_select was changed to include detach_all you need to call linux_attach_new_process directly to solve this issue or create a new function to handle forked processes.

Reply:
I tested what you said and you are right. Besides, what I said is totally wrong, since `dbg->pid` is already set to the forked `pid` before doing `r_debug_select`, so it will attach the forked process as a new process and detaching will not happen. I actually forgot the actual reason I commented this out, but I know that it will cause some ptrace calls to fail and printing errors in a specific situation. I will just revert this back to what it was before then, since it does not affect my PR in any way. After all, thanks for pointing this out as well.